File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(a, b):
    return [i for i in range(a, b+1)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc())
    else:
        return sum(inpFunc())/len(inpFunc())

a = int(input("Enter the starting integer (a): "))
b = int(input("Enter the ending integer (b): "))

print(applicatorFunc(listFunc, 's'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc = [1..5] 

--Define applicatorFunc
applicatorFunc inpFunc s = if s=='s' then sum inpFunc else (sum inpFunc)/5  

main = do
    putStrLn "Enter the starting integer (a): "
    a <- readLn
    putStrLn "Enter the ending integer (b): "
    b <- readLn
    let inpFunc = [a..b] -- Generate the list of integers from a to b
        result = applicatorFunc inpFunc 'a' -- Call applicatorFunc with inpFunc and 'a' as args
    putStrLn $ "Result = " ++ show result